tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -575,6 +576,110 @@ public:
+      }
+   }
+
+   void Initialize() {}
+   void InitTask(TTreeReader *, unsigned int) {}
+
+   // case: all types are container types
+   template <
+      typename X, typename Y, typename EXL, typename EXH, typename EYL, typename EYH,
+      std::enable_if_t<IsDataContainer<X>::value && IsDataContainer<Y>::value && IsDataContainer<EXL>::value &&
+                          IsDataContainer<EXH>::value && IsDataContainer<EYL>::value && IsDataContainer<EYH>::value,
+                       int> = 0>
+   void
+   Exec(unsigned int slot, const X &xs, const Y &ys, const EXL &exls, const EXH &exhs, const EYL &eyls, const EYH &eyhs)
+   {
+      if ((xs.size() != ys.size()) || (xs.size() != exls.size()) || (xs.size() != exhs.size()) ||
+          (xs.size() != eyls.size()) || (xs.size() != eyhs.size())) {
+         throw std::runtime_error("Cannot fill GraphAsymmErrors with values in containers of different sizes.");
+      }
+      auto thisSlotG = fGraphAsymmErrors[slot];
+      auto xsIt = std::begin(xs);
+      auto ysIt = std::begin(ys);
+      auto exlsIt = std::begin(exls);
+      auto exhsIt = std::begin(exhs);
+      auto eylsIt = std::begin(eyls);
+      auto eyhsIt = std::begin(eyhs);
+      while (xsIt != std::end(xs)) {
+         const auto n = thisSlotG->GetN(); // create only one point per iteration, but not 2
+         thisSlotG->SetPoint(n, *xsIt++, *ysIt++);
+         thisSlotG->SetPointError(n, *exlsIt++, *exhsIt++, *eylsIt++, *eyhsIt++);
+      }
+   }
+
+   // case: all types are non-container types, e.g. scalars
+   template <
+      typename X, typename Y, typename EXL, typename EXH, typename EYL, typename EYH,
+      std::enable_if_t<!IsDataContainer<X>::value && !IsDataContainer<Y>::value && !IsDataContainer<EXL>::value &&
+                          !IsDataContainer<EXH>::value && !IsDataContainer<EYL>::value && !IsDataContainer<EYH>::value,
+                       int> = 0>
+   void Exec(unsigned int slot, X x, Y y, EXL exl, EXH exh, EYL eyl, EYH eyh)
+   {
+      auto thisSlotG = fGraphAsymmErrors[slot];
+      thisSlotG->SetPoint(0, x, y);
+      thisSlotG->SetPointError(0, exl, exh, eyl, eyh);
+   }
+
+   // case: types are combination of containers and non-containers
+   // "if" condition checks that all types are neither all containers, nor all non-containers
+   template <typename X, typename Y, typename EXL, typename EXH, typename EYL, typename EYH,
+             std::enable_if_t<
+                !(!IsDataContainer<X>::value && !IsDataContainer<Y>::value && !IsDataContainer<EXL>::value &&
+                  !IsDataContainer<EXH>::value && !IsDataContainer<EYL>::value && !IsDataContainer<EYH>::value) &&
+                   !(IsDataContainer<X>::value && IsDataContainer<Y>::value && IsDataContainer<EXL>::value &&
+                     IsDataContainer<EXH>::value && IsDataContainer<EYL>::value && IsDataContainer<EYH>::value),
+                int> = 0>
+   void Exec(unsigned int, X, Y, EXL, EXH, EYL, EYH)
```

Comment:
Maybe easier to read:

```suggestion
   // case: types are combination of containers and non-containers
   // this is not supported, error out
   template <typename X, typename Y, typename EXL, typename EXH, typename EYL, typename EYH,
                    typename ...ExtraArgsToLowerPriority>
   void Exec(unsigned int, X, Y, EXL, EXH, EYL, EYH, ExtraArgsToLowerPriority...)
```

see also https://godbolt.org/z/1nejKzeed

Reply:
agreed, tested and applied